src/library-authoring/component-info/ComponentUsage.tsx

```diff
@@ -0,0 +1,94 @@
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { Collapsible } from '@openedx/paragon';
+import { Link } from 'react-router-dom';
+
+import AlertError from '../../generic/alert-error';
+import Loading from '../../generic/Loading';
+import { useFetchIndexDocuments, type ContentHit } from '../../search-manager';
+import { useComponentDownstreamLinks } from '../data/apiHooks';
+import messages from './messages';
+
+interface ComponentUsageProps {
+  usageKey: string;
+}
+
+type ComponentUsageTree = Record<string, {
+  key: string,
+  contextName: string,
+  links: ContentHit['breadcrumbs'][1][],
+}>;
+
+export const ComponentUsage = ({ usageKey }: ComponentUsageProps) => {
+  const {
+    data: dataDownstreamLinks,
+    isError: isErrorDownstreamLinks,
+    error: errorDownstreamLinks,
+    isLoading: isLoadingDownstreamLinks,
+  } = useComponentDownstreamLinks(usageKey);
+
+  const downstreamKeys = dataDownstreamLinks || [];
+
+  const {
+    data: downstreamHits,
+    isError: isErrorIndexDocuments,
+    error: errorIndexDocuments,
+    isLoading: isLoadingIndexDocuments,
+  } = useFetchIndexDocuments({
+    filter: [`usage_key IN ["${downstreamKeys.join('","')}"]`],
+    limit: downstreamKeys.length,
+    attributesToRetrieve: ['usage_key', 'breadcrumbs', 'context_key'],
+    enabled: !!downstreamKeys.length,
+  });
+
+  if (isErrorDownstreamLinks || isErrorIndexDocuments) {
+    return <AlertError error={errorDownstreamLinks || errorIndexDocuments} />;
+  }
+
+  if (isLoadingDownstreamLinks || isLoadingIndexDocuments) {
+    return <Loading />;
+  }
+
+  if (!downstreamKeys.length) {
+    return <FormattedMessage {...messages.detailsTabUsageEmpty} />;
+  }
+
+  const componentUsage = downstreamHits.reduce<ComponentUsageTree>((acc, hit) => {
+    const link = hit.breadcrumbs.at(-1);
```

Comment:
You'll need to build url to link to the units, like [here](https://github.com/openedx/frontend-app-course-authoring/blob/0c35fe8fe9e1a0998bd28f763351508718858948/src/course-libraries/CourseLibraries.tsx#L62-L69). Currently, I just see the unit names and links don't work.

![image](https://github.com/user-attachments/assets/c358eb47-da08-4a89-ac81-a394b9ef678e)

Reply:
Scratch this, I missed to reindex and test. Will come back to this.

Reply:
I think building the url here is simple enough to avoid adding the link data to index. See https://github.com/openedx/edx-platform/pull/36253#discussion_r1964789966

Reply:
Thanks @navinkarkera! Fixed!

Reply:
You're right! I may have done something wrong with my testing.
It is respecting the waffle flag as expected.